mlflow/paddle/__init__.py

```diff
@@ -254,7 +270,24 @@ def load_model(model_uri):
     local_model_path = _download_artifact_from_uri(artifact_uri=model_uri)
     flavor_conf = _get_flavor_configuration(model_path=local_model_path, flavor_name=FLAVOR_NAME)
     pd_model_artifacts_path = os.path.join(local_model_path, flavor_conf["pickled_model"])
-    return paddle.jit.load(pd_model_artifacts_path)
+    if not model:
+        return paddle.jit.load(pd_model_artifacts_path)
+    elif not isinstance(model, paddle.Model):
+        raise TypeError("Argument 'model' should be a paddle.Model, if it is not None")
+    else:
+        support_retrain = _check_if_model_supports_retrain(local_model_path)
+        if not support_retrain:
+            raise TypeError(
+                "The model doee not support re-train. \
+                Please set flag training=True when the paddle model is saved"
+            )
+        else:
+            model.load(
+                pd_model_artifacts_path,
+                skip_mismatch=skip_mismatch,
+                reset_optimizer=reset_optimizer,
+            )
+            return model
```

Comment:
```suggestion
        support_retrain = _check_if_model_supports_retrain(local_model_path)
        if not support_retrain:
            raise TypeError(
                "The model doee not support re-train. \
                Please set flag training=True when the paddle model is saved"
            )
        else:
            model.load(
                pd_model_artifacts_path,
                skip_mismatch=skip_mismatch,
                reset_optimizer=reset_optimizer,
            )
            return model
```

`model.load` works for a model logged with `training=False`, right?

Reply:
No, if we are using high level api and set ```training=False```. i.e ```model.save(..., training=False)```, then the model will be saved without any optimizer and we can just use low level api ```paddle.jit.load``` to load that model.

Reply:
Got it, thanks for the explanation :)